Pessimistic lock

angelo-rendina-prima opened the pull request.

cottinisimone commented on examples/optimistic_lock/Cargo.toml:
```diff
@@ -0,0 +1,23 @@
+[package]
+authors = ["Oliver Browne <oliver.browne@prima.it"]
```
lol you can write your name :)

cottinisimone commented on examples/optimistic_lock/Cargo.toml:
```diff
@@ -0,0 +1,23 @@
+[package]
+authors = ["Oliver Browne <oliver.browne@prima.it"]
+description = "Simple esrs side-effect example"
```
Description?

cottinisimone commented on src/esrs/postgres/store.rs:
```diff
@@ -201,6 +213,20 @@ where
+        let lock_guard = PgStoreLockGuardAsyncSendBuilder {
+            lock: PgAdvisoryLock::with_key(PgAdvisoryLockKey::BigInt(key as i64)),
+            guard_builder: |lock: &PgAdvisoryLock| {
+                Box::pin(async move { lock.acquire(connection).await.expect("TODO") })
```
TODO :)

cottinisimone commented on Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM public.ecr.aws/prima/rust:1.62.1-1
+FROM public.ecr.aws/prima/rust:1.64.0
```
Maybe now there's the1.65?

angelo-rendina-prima replied:
Nope still 1.64

angelo-rendina-prima replied:
Actually just bumped 🚀

cottinisimone commented on examples/pessimistic_lock/src/aggregate.rs:
```diff
@@ -0,0 +1,68 @@
+use serde::{Deserialize, Serialize};
```
I think you can include another example, so you can avoid rewriting this aggregate stuff

angelo-rendina-prima replied:
Eh all other examples have their own aggregate, I just went with the slimmest I could think of (avoiding projectors and policies).

angelo-rendina-prima replied:
Done, recycled the aggregate from another example.

miterst reviewed (approved):
LGTM. We can give it a try!

angelo-rendina-prima commented:
Bumped Rust image to 1.65, had to make a tiny change around deref to satisfy the new version of clippy.
Renamed the example to `locking_strategies`, and added two extra snippets about atomic **reads**.
